packages/adyen-integration/src/adyenv3/AdyenV3PaymentMethod.tsx

```diff
@@ -0,0 +1,206 @@
+import {
+    AdyenV3CreditCardComponentOptions,
+    AdyenV3ValidationState,
+    CardInstrument,
+    PaymentInitializeOptions,
+} from '@bigcommerce/checkout-sdk';
+import React, { FunctionComponent, useCallback, useRef, useState } from 'react';
+
+import { HostedWidgetComponentProps } from '@bigcommerce/checkout/hosted-widget-integration';
+import { LocaleProvider } from '@bigcommerce/checkout/locale';
+import {
+    CheckoutContext,
+    PaymentFormContext,
+    PaymentMethodProps,
+    PaymentMethodResolveId,
+    toResolvableComponent,
+} from '@bigcommerce/checkout/payment-integration-api';
+import { FormContext, LoadingOverlay } from '@bigcommerce/checkout/ui';
+
+import AdyenV3CardValidation from './AdyenV3CardValidation';
+import AdyenV3Form from './AdyenV3Form';
+
+export interface AdyenOptions {
+    [key: string]: AdyenV3CreditCardComponentOptions;
+}
+
+export enum AdyenV3PaymentMethodType {
+    scheme = 'scheme',
+    bcmc = 'bcmc',
+}
+
+interface AdyenPaymentMethodRef {
+    shouldShowModal: boolean;
+    shouldShowNumberField?: boolean;
+    cancelAdditionalAction?(): void;
+}
+
+const AdyenV3PaymentMethod: FunctionComponent<PaymentMethodProps> = ({
+    checkoutService,
+    checkoutState,
+    paymentForm,
+    method,
+    language,
+    onUnhandledError,
+    ...rest
+}) => {
+    const ref = useRef<AdyenPaymentMethodRef>({
+        shouldShowModal: true,
+    });
+
+    const [showAdditionalActionContent, setShowAdditionalActionContent] = useState<boolean>(false);
+    const [cardValidationState, setCardValidationState] = useState<AdyenV3ValidationState>();
+    const containerId = `adyen-${method.id}-component-field`;
+    const additionalActionContainerId = `adyen-${method.id}-additional-action-component-field`;
+    const cardVerificationContainerId = `adyen-${method.id}-tsv-component-field`;
+    const component = method.id;
+    const shouldHideInstrumentExpiryDate = component === AdyenV3PaymentMethodType.bcmc;
+    const adyenOptions: AdyenOptions = {
+        [AdyenV3PaymentMethodType.scheme]: {
+            hasHolderName: true,
+            holderNameRequired: true,
+        },
+    };
+
+    const onBeforeLoad = useCallback((shopperInteraction: boolean) => {
+        ref.current.shouldShowModal = shopperInteraction;
+
+        if (ref.current.shouldShowModal) {
+            setShowAdditionalActionContent(true);
+        } else {
+            setShowAdditionalActionContent(false);
+        }
```

Comment:
This code can be simplified:

```suggestion
            setShowAdditionalActionContent(ref.current.shouldShowModal);
```